tests/deploy/e2e/test_autostarted.py

```diff
@@ -1344,3 +1344,51 @@ async def wait_for_down_status() -> bool:
     assert result.code == 200
     summary = result.result["metadata"]["deploy_summary"]
     assert summary["by_state"]["available"] == 1, f"Unexpected summary: {summary}"
+
+
+async def test_rps_state_deploying(
+    snippetcompiler,
+    server,
+    ensure_resource_tracker_is_started,
+    client,
+    clienthelper,
+    environment,
+    async_finalizer,
+    tmp_path,
+):
+    """
+    Verify that the is_deploying flag is correctly set when deploying starts and finishes
+    """
+
+    file_to_remove = tmp_path / "file"
+    file_to_remove.touch()
+
+    config.Config.set("config", "environment", environment)
+
+    model = f"""
+import minimalwaitingmodule
+minimalwaitingmodule::WaitForFileRemoval(name="test_sleep", agent="agent1", path="{file_to_remove}")
+"""
+    snippetcompiler.setup_for_snippet(model, ministd=True, index_url="https://pypi.org/simple")
```

Comment:
```suggestion
    snippetcompiler.setup_for_snippet(model, ministd=True)
```
I don't think this is required. Do we do it anywhere else?

Reply:
Yes, I copied it from the test case above

Reply:
I see. I still don't think it's required, and I'd prefer to see it go (unless I'm mistaken). But I won't hold back the PR for this.

Reply:
Removing the index_url causes this call `await wait_for_resources_in_state(client, uuid.UUID(environment), nr_of_resources=1, state=const.ResourceState.deploying)` to always timeout.

Reply:
Ah I see, we need it on the agent side as well. Sorry for the noise.